src/inmanta/deploy/scheduler.py

```diff
@@ -628,6 +636,37 @@ async def report_resource_state(
                             deploying=set(),
                         )
 
+    async def get_last_non_deploying_state_for_dependencies(
+        self, resource_id: ResourceIdStr
+    ) -> dict[ResourceIdStr, const.ResourceState]:
+        """
+        Get resource state for every dependency of a given resource from the scheduler state.
+        The state is then converted to const.ResourceState.
+
+        Acquires the appropriate locks
+
+        :param resource_id: The id of the resource to find the dependencies for
+        """
+        async with self._scheduler_lock:
+            requires_view: Mapping[ResourceIdStr, Set[ResourceIdStr]] = self._state.requires.requires_view()
```

Comment:
Are you aware that `requires_view()` simply returns `self`? i.e. I don't think there's a need for this, you can simply use `self._state.requires` directly, unless this was a conscious decision?

Reply:
Yes, I was aware. I find it more legible like this, but no strong opinion either way